spec/features/find/search/undergraduate/course_results_spec.rb

```diff
@@ -0,0 +1,389 @@
+    given_i_have_2025_courses
+    and_i_am_in_the_2025_cycle
+    and_the_tda_feature_flag_is_active
+    when_i_visit_the_start_page
+    and_i_select_the_across_england_radio_button
+    and_i_click_continue
+    and_i_choose_secondary
+    and_i_click_continue
+    and_i_choose_subjects
+    and_i_click_continue
+    then_i_am_on_the_undergraduate_question_page
+
+    when_i_choose_yes_i_have_a_degree
+    and_i_click_continue
+    then_i_am_on_the_visa_status_page
+    and_the_back_link_points_to_the_degree_question
+
+    when_i_choose_no
+    and_i_click_to_find_courses
+    then_i_am_on_results_page
+    and_all_filters_are_visible
+    and_i_can_see_only_postgraduate_courses
+  end
+
+  def given_i_have_2025_courses
+    _, provider = setup_recruitment_cycle(year: 2025)
+
+    @biology_course = create(:course, :published_teacher_degree_apprenticeship, :secondary, provider:, name: 'Biology', subjects: [find_or_create(:secondary_subject, :biology)])
+    @history_course = create(:course, :published_teacher_degree_apprenticeship, :secondary, provider:, name: 'History', subjects: [find_or_create(:secondary_subject, :history)])
+    @primary_with_science_course = create(:course, :published_teacher_degree_apprenticeship, :primary, provider:, name: 'Primary with science', subjects: [find_or_create(:primary_subject, :primary_with_science)])
+
+    @mathematics_course = create(:course, :published_postgraduate, :secondary, provider:, name: 'Mathematics', subjects: [find_or_create(:secondary_subject, :mathematics)])
+    @chemistry_course = create(:course, :published_postgraduate, :secondary, provider:, name: 'Chemistry', subjects: [find_or_create(:secondary_subject, :chemistry)])
+  end
+
+  def setup_recruitment_cycle(year:)
+    recruitment_cycle = create(:recruitment_cycle, year:)
+    user = create(:user, providers: [build(:provider, recruitment_cycle:, provider_type: 'lead_school', sites: [build(:site), build(:site)], study_sites: [build(:site, :study_site), build(:site, :study_site)])])
+    provider = user.providers.first
+    create(:provider, :accredited_provider, provider_code: '1BJ')
+    accredited_provider = create(:provider, :accredited_provider, provider_code: '1BJ', recruitment_cycle:)
+    provider.accrediting_provider_enrichments = []
+    provider.accrediting_provider_enrichments << AccreditingProviderEnrichment.new(
+      {
+        UcasProviderCode: accredited_provider.provider_code,
+        Description: 'description'
+      }
+    )
+    [recruitment_cycle, provider]
+  end
+
+  def given_i_have_2024_courses
+    _, provider = setup_recruitment_cycle(year: 2024)
+
+    create(:course, :resulting_in_pgce_with_qts, provider:, name: 'Chemistry')
+    create(:course, :resulting_in_pgce_with_qts, provider:, name: 'Mathematics')
+  end
+
+  def and_i_am_in_the_2025_cycle
+    Timecop.travel(Time.zone.local(2024, 10, 1, 9, 1)) # after Find opens
```

Comment:
I agree, but Find/Pub is different from apply where we could pass `CycleTimetable.apply_opens(CycleTimetable.next_year) + 1.day`

Also I'd like to be specific for the 2025 cycle.

But there is no way to do it easily:

```
Timecop.travel(CycleTimetable.find_opens(2025))
```

I could change the method but the implications was so large for a fundamental code to be needed in a spec that I decided not to change it.

I am happy to hear alternatives.

Reply:
This definitely is not a blocker but It feels wrong to hardcode dates, especially such specific ones. 

Would this work? 

```
  def and_i_am_in_the_2025_cycle
    Timecop.travel(Find::CycleTimetable.find_reopens)
    allow(Settings).to receive(:current_recruitment_cycle_year).and_return(2025)
  end

  def and_i_am_in_the_2024_cycle
    Timecop.travel(Find::CycleTimetable.find_opens)
    allow(Settings).to receive(:current_recruitment_cycle_year).and_return(2024)
  end
```

I assume the only reason you even need this is so that this doesn't fail when Find is closed because it currently passes without it.